### PR TITLE
BEATs checkpoint/config fixes + regressions

### PR DESCRIPTION
**Pull request by nkundiushuti**
## Summary
- Fix BEATs checkpoint loading across HF safetensors + GCS `.pt` formats
- Add `ModelSpec.init_config` to carry BEATsConfig for safetensors-based official models
- Add BEATs regression coverage (probe + sentinel tensor hashes)
- Skip private-GCS checkpoint equivalence test when unauthenticated

## Test plan
- [x] `pytest tests/integration/test_official_models_regression.py -k beats`
**Goal**: catch regressions in how official_models/*.yml entries load and behave.
**What it does:**
Loads official registry models (including the 3 BEATs ones) via the normal load_model() path.
Runs a tiny linear-probe smoke check to ensure embeddings are usable (loss decreases).
For BEATs specifically, it also compares against pinned fixtures:
Pinned probe losses (initial + final) to detect changes in forward path / determinism.
Pinned config + sentinel tensor hashes for a handful of backbone tensors, to detect “weights didn’t load” or “different checkpoint/config got used” failures.
- [x] `pytest tests/unittests/test_official_beats_sdpa_equivalence.py`
**Goal**: ensure the BEATs attention implementation change (manual attention vs PyTorch SDPA/Flash attention path) is numerically equivalent.
**What it does:**
Attention-layer unit check: constructs a legacy attention module and the SDPA-based one with identical weights and asserts outputs match within tolerances.
End-to-end checkpoint check: loads the BEATs SSL .pt checkpoint, builds two full BEATs models:
one using SDPA attention
one swapped to legacy attention
runs the same random audio through both and asserts the max output difference is below a threshold.
If the checkpoint is in a private GCS bucket and can’t be accessed (401/403), it skips instead of failing CI.

### CODE REVIEW

File: avex/models/beats/beats.py [hunk 1]
```diff
@@ -61,7 +62,7 @@ class BEATsConfig(BaseModel):
     # Training dynamics
     layer_wise_gradient_decay_ratio: float = Field(1.0, description="Ratio for layer-wise gradient decay")
     layer_norm_first: bool = Field(False, description="Apply layernorm first in the transformer")
-    deep_norm: bool = Field(False, description="Apply deep_norm first in the transformer")
+    deep_norm: bool = Field(True, description="Apply deep_norm first in the transformer")
```
Comment (GaganNarula) [inline avex/models/beats/beats.py:65]: these values of `deep_norm=True` and `finetuned_model=False` match the pretrained microsoft beats right ?

Comment (GaganNarula) [reply avex/models/beats/beats.py:65]: ok yes i see it in the yaml , ignore the comment

File: avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+# BEATs iter3+AS2M SSL pretrained backbone (GCS source-of-truth).
+#
+# This file is intentionally NOT part of `official_models` (i.e. it is not
+# auto-registered as a user-facing model). It exists solely to provide a
+# schema-validated place to store checkpoint URIs used internally by
+# `avex.models.beats_model`.
+checkpoint_path: gs://representation-learning/pretrained/BEATs_iter3_plus_AS2M.pt
```
Comment (GaganNarula) [inline avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml:8]: how is this path used ? i'm just checking whether this path isnt accidentally used in a call to load the checkpoint

File: avex/models/beats_model.py [hunk 3]
```diff
@@ -13,37 +13,58 @@
...
+        return checkpoint_path
+
+    if fine_tuned:
+        yaml_data = load_packaged_yaml_mapping(
+            package=_CHECKPOINT_CONFIGS_PKG, name="beats_iter3_plus_as2m_finetuned_cpt2"
+        )
+        checkpoint_path = yaml_data.get("checkpoint_path")
+        if not (isinstance(checkpoint_path, str) and checkpoint_path):
+            raise ValueError(
+                "No checkpoint_path found in packaged config for BEATs finetuned "
```
Comment (GaganNarula) [inline avex/models/beats_model.py:57]: just reiterating this from the last PR.. why should these checks every fail ?

File: avex/models/beats_model.py [hunk 4]
```diff
@@ -108,38 +130,62 @@ def __init__(
...
+                    cfg_name = "beats_iter3_plus_as2m_finetuned_cpt2" if fine_tuned else "beats_iter3_plus_as2m_ssl"
+                    beats_cfg_raw = load_packaged_yaml_mapping(package=_CHECKPOINT_CONFIGS_PKG, name=cfg_name).get(
+                        "beats_cfg"
+                    )
+                    if not isinstance(beats_cfg_raw, dict):
+                        raise ValueError(f"Missing/invalid beats_cfg in packaged YAML: {cfg_name}.yml")
+                    cfg = BEATsConfig(**beats_cfg_raw)
+                    if use_naturelm:
+                        # NatureLM uses BEATs weights but expects the fine-tuned model mode.
+                        cfg = cfg.model_copy(update={"finetuned_model": True})
```
Comment (GaganNarula) [inline avex/models/beats_model.py:184]: this whole loading bit is quite confusing ... when is _load_checkpoint called ?

File: avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml (hunk 2 above)
Comment (nkundiushuti) [reply avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml:8]: this path is used solely in beats_model.py because of this pretrained=True parameter which triggers loading these microsoft beats checkpoints (which are not public anywhere). those are not official models so I added their configs here (we need separate configs for pretrained ssl vs pretrained finetuned)

File: avex/models/beats_model.py (hunk 3 above)
Comment (nkundiushuti) [reply avex/models/beats_model.py:57]: if someone mingles with these yaml in an editable install, we need to raise an issue here.

File: avex/models/beats_model.py (hunk 4 above)
Comment (nkundiushuti) [reply avex/models/beats_model.py:184]: this is only to build the model. injecting the weights happens in the generic loader in avex/models/utils/load.py when checkpoint_path is present

**Review by GaganNarula — APPROVED** [review]
i'm not 100% sure i can really follow everything but if the tests are covering the new fix then i'm ok

File: avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml (hunk 2 above)
Comment (GaganNarula) [reply avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml:8]: ok but will a user who does not have gcs access be able to load the model when pretrained=True ?  thats what we need to make sure

Comment (nkundiushuti) [reply avex/api/configs/checkpoints/beats_iter3_plus_as2m_ssl.yml:8]: No, they can't, this is purely for internal purposes eg the what matters recipes
